nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,149 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+  clightConf = pkgs.writeText "clight.conf" cfg.extraConfig;
+in {
+  options.services.clight = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable clight or not.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clight;
+      defaultText = "pkgs.clight";
+      description = ''
+        Clight derivation to use.
+      '';
+    };
+
+    clightd = mkOption {
+      type = types.package;
+      default = pkgs.clightd;
+      defaultText = "pkgs.clightd";
+      description = ''
+        Clightd derivation to use.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Additional configuration to pass to clight.conf.
+      '';
+    };
+
+    extraOptions = mkOption {
```

Comment:
I think the approach I describe in NixOS/rfcs#42 is a perfect fit for here: Instead of having these stringly-typed `extra*` options, have a `settings` option of a type like `attrsOf (nullOr (either str (either int (listOf float)))` (and whatever else can go in the config file), then convert that option to the config file string.

This then allows users to override settings and have them work in a composable way. It also seems that you can specify everything in the configuration file, so command line options aren't needed (and would only make it more complicated).

Reply:
I'm not entirely sure I interpreted this correctly but I think I did it?